diff --git a/crates/gkr/src/lib.rs b/crates/gkr/src/lib.rs
index 02e6e94..f40c572 100644
--- a/crates/gkr/src/lib.rs
+++ b/crates/gkr/src/lib.rs
@@ -59,14 +59,14 @@ fn prove_layer(ps: &mut ProverState, point: Point, mut wnext: Vec<Field>) -> (Po
         let (lo_r, hi_r) = mle_r.split_at_mut(h);
 
         let (sum_0, sum_inf) = lo_l
-            .par_iter_mut()
-            .zip(lo_r.par_iter_mut())
-            .zip(hi_l.par_iter())
-            .zip(hi_r.par_iter())
-            .zip(eq.par_iter())
-            .with_min_len(PARALLEL_MIN_LANES)
+            .iter_mut()
+            .zip(lo_r.iter_mut())
+            .zip(hi_l.iter())
+            .zip(hi_r.iter())
+            .zip(eq.iter())
+            // .with_min_len(PARALLEL_MIN_LANES)
             .fold(
-                || (Wide256::zero(), Wide256::zero()),
+                (Wide256::zero(), Wide256::zero()),
                 |(mut sum_0, mut sum_inf), ((((l_lo, r_lo), &l_hi), &r_hi), &e)| {
                     let (d_l, d_r) = (l_hi - *l_lo, r_hi - *r_lo);
 
@@ -78,22 +78,22 @@ fn prove_layer(ps: &mut ProverState, point: Point, mut wnext: Vec<Field>) -> (Po
 
                     (sum_0, sum_inf)
                 },
-            )
-            .reduce(
-                || (Wide256::zero(), Wide256::zero()),
-                |(a0, ainf), (b0, binf)| (a0 + b0, ainf + binf),
             );
+        // .reduce(
+        //     || (Wide256::zero(), Wide256::zero()),
+        //     |(a0, ainf), (b0, binf)| (a0 + b0, ainf + binf),
+        // );
 
         ps.prover_message(&[factor * sum_0.reduce(), factor * sum_inf.reduce()]);
 
         let r = ps.verifier_message();
         next_point.push_back(r);
 
-        lo_l.par_iter_mut()
-            .zip(lo_r.par_iter_mut())
-            .zip(hi_l.par_iter())
-            .zip(hi_r.par_iter())
-            .with_min_len(PARALLEL_MIN_LANES)
+        lo_l.iter_mut()
+            .zip(lo_r.iter_mut())
+            .zip(hi_l.iter())
+            .zip(hi_r.iter())
+            // .with_min_len(PARALLEL_MIN_LANES)
             .for_each(|(((l_lo, r_lo), &l_hi), &r_hi)| {
                 // Recalculating delta rather than writing it in the top half of the array in the previous loop to save on writes.
                 // No benchmarking has been done to check the difference.
